velox/exec/Task.cpp

```diff
@@ -386,11 +386,13 @@ void Task::start(
           !self->hasPartitionedOutput_,
           "Only one output pipeline per task is supported");
       self->hasPartitionedOutput_ = true;
-      bufferManager->initializeTask(
-          self,
-          partitionedOutputNode->isBroadcast(),
-          partitionedOutputNode->numPartitions(),
-          self->numDriversInPartitionedOutput_ * numSplitGroups);
+      if (partitionedOutputNode->isStreaming()) {
```

Comment:
I believe self->hasPartitionedOutput_ needs to be false if we are no going to use POBM. E.g. hasPartitionedOutput_ needs to go together with bufferManager->initializeTask.

Reply:
Sounds good. So an streaming or batch shuffle has the choice to use hasPartitionedOutput or not